src/components/TextInput/BaseTextInput/index.tsx

```diff
@@ -281,48 +320,49 @@ function BaseTextInput(props) {
                                 {isMultiline && <View style={[styles.textInputLabelBackground, styles.pointerEventsNone]} />}
                                 <TextInputLabel
                                     isLabelActive={isLabelActive.current}
-                                    label={props.label}
+                                    label={label ?? ''}
                                     labelTranslateY={labelTranslateY}
                                     labelScale={labelScale}
-                                    for={props.nativeID}
+                                    for={inputProps.nativeID}
                                 />
                             </>
                         ) : null}
                         <View style={[styles.textInputAndIconContainer, isMultiline && hasLabel && styles.textInputMultilineContainer, styles.pointerEventsBoxNone]}>
-                            {Boolean(props.prefixCharacter) && (
+                            {Boolean(prefixCharacter) && (
                                 <View style={styles.textInputPrefixWrapper}>
                                     <Text
                                         tabIndex={-1}
                                         style={[styles.textInputPrefix, !hasLabel && styles.pv0, styles.pointerEventsNone]}
                                         dataSet={{[CONST.SELECTION_SCRAPER_HIDDEN_ELEMENT]: true}}
                                     >
-                                        {props.prefixCharacter}
+                                        {prefixCharacter}
                                     </Text>
                                 </View>
                             )}
                             <RNTextInput
-                                ref={(ref) => {
-                                    if (typeof props.innerRef === 'function') {
-                                        props.innerRef(ref);
-                                    } else if (props.innerRef && _.has(props.innerRef, 'current')) {
+                                ref={(element) => {
+                                    if (typeof ref === 'function') {
+                                        ref(element);
+                                    } else if (ref && 'current' in ref) {
                                         // eslint-disable-next-line no-param-reassign
-                                        props.innerRef.current = ref;
+                                        ref.current = element;
                                     }
-                                    input.current = ref;
+                                    // @ts-expect-error We need to reassign this ref to the input ref
+                                    input.current = element;
```

Comment:
It's possible to get rid of @ts-expect-error , for example
`(input.current as React.Component<AnimatedProps<TextInputProps>> | null) = element;`

`React.Component<AnimatedProps<TextInputProps>>` is actually a type of `AnimatedTextInput`. So maybe you can even improve it/update ref typing